front/src/applications/editor/tools/rangeEdition/tool-factory.tsx

```diff
@@ -80,50 +82,72 @@ function getRangeEditionTool<T extends EditorRange>({
-              },
-              onClick({ setState, state: { initialEntity } }) {
-                setState({
-                  entity: cloneDeep(initialEntity),
-                });
-              },
-            }
-          : {
-              id: 'reset-catenary',
-              icon: BiReset,
-              labelTranslationKey: 'Editor.tools.catenary-edition.actions.reset-catenary',
-              isDisabled({ state: { entity, initialEntity } }) {
-                return isEqual(entity, initialEntity);
-              },
-              onClick({ setState, state: { initialEntity } }) {
-                setState({
-                  entity: cloneDeep(initialEntity),
-                });
-              },
-            },
+        {
+          id: `new-${getObjTypeAction(layersEntity.objType)}`,
+          icon: IoMdAddCircleOutline,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.new-${getObjTypeAction(layersEntity.objType)}`,
+          onClick({ setState }) {
+            setState(getInitialState());
+          },
+        },
+        {
+          id: `reset-${getObjTypeAction(layersEntity.objType)}`,
+          icon: BiReset,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.reset-${getObjTypeAction(layersEntity.objType)}`,
+          isDisabled({ state: { entity, initialEntity } }) {
+            return isEqual(entity, initialEntity);
+          },
+          onClick({ setState, state: { initialEntity } }) {
+            setState({
+              entity: cloneDeep(initialEntity),
+            });
+          },
+        },
+      ],
+      [
+        {
+          id: 'delete-entity',
+          icon: BsTrash,
+          labelTranslationKey: `Editor.tools.${getObjTypeEdition(
+            layersEntity.objType
+          )}-edition.actions.delete-${getObjTypeAction(layersEntity.objType)}`,
+          // Show button only if we are editing
+          isDisabled({ state }) {
+            return state.initialEntity.geometry.coordinates.length === 0;
+          },
+          onClick({ infraID, openModal, closeModal, forceRender, state, setState, dispatch, t }) {
+            openModal(
+              <ConfirmModal
+                title={t(
+                  `Editor.tools.${getObjTypeEdition(
+                    layersEntity.objType
+                  )}-edition.actions.delete-${getObjTypeAction(layersEntity.objType)}`
+                )}
+                onConfirm={async () => {
+                  await dispatch<ReturnType<typeof save>>(
+                    // We have to put state.initialEntity in array because delete initially works with selection which can get multiple elements
+                    save(infraID, { delete: [state.initialEntity] })
+                  );
+                  setState(getInitialState());
+                  closeModal();
+                  forceRender();
```

Comment:
I wonder whether it wouldn't be better to put this forceRender in the save directly, since we cannot await

Reply:
When submitting a new object, we also use the save function but without using forceRender. So if we add it instead save, it would result with a unnecessary rerender.